pkgs/by-name/ka/kanidmWithSecretProvisioning/package.nix

```diff
@@ -0,0 +1,42 @@
+# If using this package, kanidm will be built with two patches allowing both
+# oauth2 basic secrets and admin credentials to be provisioned.
+# This is NOT officially supported (and will likely never be),
+# see https://github.com/kanidm/kanidm/issues/1747.
+# Please report any provisioning-related errors to
+# https://github.com/oddlama/kanidm-provision/issues/ instead.
+{
+  kanidm,
+  lib,
+  nixosTests,
+}:
+kanidm.overrideAttrs (old: {
```

Comment:
As per #421201, introducing new usages of `.overrideAttrs` within `nixpkgs` is probably a bad idea.

Reply:
If you would like to split `kanidm` from `kanidmWithSecretProvisioning`, then you should duplicate all the code.

Think about it that way: You don't appear to want to be involved in `kanidmWithSecretProvisioning` anymore. But whenever you bump `kanidm`, there is a high change you regress `kanidmWithSecretProvisioning`. For example a patch may no longer apply and needs some work. By properly decoupling and duplicating all of this, you are able to move `kanidm` truly independently of `kanidmWithSecretProvisioning`.

If you do want to keep them tied together, but maybe just a bit less than it was up until now, then you would establish a proper internal `.override` interface, which also doubles as a sort of contract between `kanidm` and `kanidmWithSecretProvisioning`.